gips/data/core.py

```diff
@@ -827,7 +831,78 @@ def products2assets(cls, products):
         return set(assets)
 
     @classmethod
+    def query_service(cls, products, tiles, textent, update=False, force=False, grouped=False):
+        """
+        query data service -- product-level interface
+        """
+        if grouped:
+            response = {}
+        else:
+            response  = []
```

Comment:
A changing return type is kind of code-smelly to me; at the very least it should be mentioned in the docstring.  Otherwise it should be taken out; just make one of the consumers do the necessary frobbing into the dict.